src/test/custom_loader_test.cpp

```diff
@@ -0,0 +1,753 @@
+        {{DUMMY_MODEL_INPUT_NAME,
+            std::tuple<ovms::shape_t, tensorflow::DataType>{{1, 10}, tensorflow::DataType::DT_FLOAT}}});
+    tensorflow::serving::PredictResponse response;
+    ASSERT_TRUE(performInferenceWithRequest(request, response) != ovms::StatusCode::OK);
+}
+
+TEST_F(TestCustomLoader, CustomLoaderSingleVersion) {
+    const char* custom_loader_config_model = R"({
+       "custom_loader_config_list":[
+         {
+          "config":{
+            "loader_name":"sample-loader",
+            "library_path": "/tmp/libsampleloader.so"
+          }
+         }
+       ],
+      "model_config_list":[
+        {
+          "config":{
+            "name":"dummy",
+            "base_path": "/ovms/src/test/dummy",
+            "nireq": 1,
+            "custom_loader_options": {"loader_name": "sample-loader", "xml_file": "dummy.xml", "bin_file": "dummy.bin"}
+          }
+        }
+      ]
+    })";
+
+    std::string fileToReload = "/tmp/ovms_config_cl.json";
+    createConfigFileWithContent(custom_loader_config_model, fileToReload);
+
+    ovms::ModelManager& manager = ovms::ModelManager::getInstance();
+    manager.startFromFile(fileToReload);
+
+    ModelServiceImpl s;
+    tensorflow::serving::GetModelStatusRequest req;
+    tensorflow::serving::GetModelStatusResponse res;
+
+    auto model_spec = req.mutable_model_spec();
+    model_spec->Clear();
+    model_spec->set_name("dummy");
+
+    ::grpc::Status ret = test_PerformModelStatusRequestX(s, req, res);
+    EXPECT_EQ(ret.ok(), true);
+}
+
+TEST_F(TestCustomLoader, CustomLoaderGetStatus) {
+    const char* custom_loader_config_model = R"({
+       "custom_loader_config_list":[
+         {
+          "config":{
+            "loader_name":"sample-loader",
+            "library_path": "/tmp/libsampleloader.so"
+          }
+         }
+       ],
+      "model_config_list":[
+        {
+          "config":{
+            "name":"dummy",
+            "base_path": "/ovms/src/test/dummy",
+            "nireq": 1,
+            "custom_loader_options": {"loader_name": "sample-loader", "xml_file": "dummy.xml", "bin_file": "dummy.bin"}
+          }
+        }
+      ]
+    })";
+
+    // std::string fileToReload = "/tmp/ovms_config_cl.json";
+    std::string fileToReload = createConfigFileWithContent(custom_loader_config_model);
+
+    ovms::ModelManager& manager = ovms::ModelManager::getInstance();
+    manager.startFromFile(fileToReload);
+
+    ModelServiceImpl s;
+    tensorflow::serving::GetModelStatusRequest req;
+    tensorflow::serving::GetModelStatusResponse res;
+
+    auto model_spec = req.mutable_model_spec();
+    model_spec->Clear();
+    model_spec->set_name("dummy");
+
+    ::grpc::Status ret = test_PerformModelStatusRequestX(s, req, res);
+    EXPECT_EQ(ret.ok(), true);
+}
+
+TEST_F(TestCustomLoader, CustomLoaderGetStatusDeleteModelGetStatus) {
+    const char* custom_loader_config_model = R"({

```

Comment:
You are reusing the same config and defining it several times.

Reply:
Changed it